### DISCUSSION TITLE
What's the advantage of using `ToolbarPrimitive.ToggleGroup` over `ToggleGroup`?

### DISCUSSION THREAD

**Discussion by funwithtriangles** [radix-ui/primitives#1277] [Help] [answered]
I'd like to use the same styled `ToggleGroup` in both my toolbar and some other parts of my app. What's the best approach here? Should I just use the base `ToggleGroup` inside of my toolbar? I'm curious to know why `ToolbarPrimitive` also exports a very similar component.

**Comment by jjenzz** [comment, marked as answer]
each radix component exports its own parts because the component might attach additional functionality or accessibility for that specific use-case. just because they look the same doesn't necessarily mean the a11y/functionality is the same unfortunately.

With `Toolbar.ToggleGroup`:


https://user-images.githubusercontent.com/175330/160438137-0e667f7d-c180-43ed-98cf-76f03207bf10.mp4


With `ToggleGroup`:


https://user-images.githubusercontent.com/175330/160438723-9334654a-a7e1-461f-b9bf-5fdb20fead9c.mp4

**Reply by funwithtriangles** [reply]
Makes a lot of sense, thanks for clearing that up! So I guess the best approach is to create some reusable base styles and apply them to both components separately.

**Reply by jjenzz** [reply]
my pleasure and yep, sharing styles is what you're after here 🙂

**Reply by funwithtriangles** [reply]
I've had one more thought on this topic and would be curious to hear your thoughts. What's the best approach if I actually wanted to use `RadioGroup` inside of my toolbar? `Toolbar` doesn't export `Toolbar.RadioGroup`

**Reply by benoitgrelard** [reply]
`RadioGroup` would usually be used in forms, hence why we went with `ToggleGroup` as a more generalised component within a toolbar. You can achieve the same result and you don't need to submit a value in a toolbar.

**Reply by funwithtriangles** [reply]
Okie doke! So I just need to make sure I'm controlling `ToggleGroup` so it acts in a radio fashion. Thanks!

**Reply by benoitgrelard** [reply]
`type="single"` gets you there, but I guess you mean ensuring you can't "unselect" a value?
Like in this example: https://www.radix-ui.com/docs/primitives/components/toggle-group#ensuring-there-is-always-a-value

**Reply by funwithtriangles** [reply]
yep makes sense, thanks!
